src/checkers/inference/model/serialization/ToStringSerializer.java

```diff
@@ -260,13 +265,9 @@ public String serialize(CombVariableSlot slot) {
     }
 
     protected String indent(String str) {
-        // create additional indentation strings up to the current level of indentation, if it
-        // doesn't exist in the indentStrings array list
-        for (int i = indentStrings.size(); i <= indent; i++) {
-            indentStrings.add(indentStrings.get(i - 1) + INDENT_STRING);
-        }
-
-        return indentStrings.get(indent) + str;
+        generateIndentations();
```

Comment:
It seems `generateIndentations()` try to encapsulate the generation logic from this `indent()` method.

However, since `indent()` method is the only client use it, why encapsulate it?

It makes code a bit harder to read as now I need to jump through methods...

It would be more clear to just embed the catching loop within the `indent` method, just like other lazy-evaluation method, we write:
```java
getObj() {
if (obj == null) new Object();
return obj;
}
```

But not:
```java
getObj() {
 if (obj == null) createNewObject();
return obj;
}

//somewhere else
void createNewObject() {
  obj = new Object();
}
```